### PR TITLE
Feature/support structured authoring

### PR DESCRIPTION
**Pull request by Roblinde**

### CODE REVIEW

File: Contentful.Core/Models/AuthoringModels.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        /// <summary>
+        /// The list of content this paragraph contains.
+        /// </summary>
+        public List<IContent> Content { get; set; }
+    }
+
+    /// <summary>
+    /// Represents a heading content node.
+    /// </summary>
+    public class Heading : IContent
```
Comment (dborsatto) [inline Contentful.Core/Models/AuthoringModels.cs]: You chose to unify the headings into 1? I was thinking about doing the same, but I was wondering about possible drawbacks

File: Contentful.Core/ContentfulClient.cs [hunk 2]
```diff
@@ -299,6 +304,11 @@ private void ResolveLinks(JObject json, JObject entryToken, ISet<string> process
 
                 var grandParent = (JObject)linkToken.Parent.Parent;
 
+                if((grandParent.Parent.Parent.Parent.Parent is JObject) && grandParent.Parent.Parent.Parent.Parent.Value<string>("nodeType") == "embedded-entry-block")
```
Comment (dborsatto) [inline Contentful.Core/ContentfulClient.cs]: 😵

File: Contentful.Core/Models/Authoring.cs [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+        private string MarkToHtmlTag(Mark mark)
+        {
+            switch (mark.Type)
+            {
+                case "bold":
+                    return "strong";
+                case "underline":
+                    return "u";
+                case "italic":
+                    return "em";
```
Comment (dborsatto) [inline Contentful.Core/Models/Authoring.cs]: I think you're missing `code`

File: Contentful.Core/Models/AuthoringModels.cs (hunk 1 above)
Comment (Roblinde) [reply Contentful.Core/Models/AuthoringModels.cs]: Yeah, I suggested doing this in the spec. The main drawback is that I need to customize the serialization of the heading to conform to the current structure.

File: Contentful.Core/ContentfulClient.cs (hunk 2 above)
Comment (Roblinde) [reply Contentful.Core/ContentfulClient.cs]: Yeah, this is what happens with uneccessary nesting... data.target.sys

File: Contentful.Core/Models/Authoring.cs (hunk 3 above)
Comment (Roblinde) [reply Contentful.Core/Models/Authoring.cs]: Probably a few more, right? I will add them once they're available in the web app.

**Pull request merged**